src/routes/users/preferences.js

```diff
@@ -52,6 +65,20 @@ router.post('/preferences', jwtAuth('write:preferences'), async (req, res, next)
       payload.creatorPitch = creatorPitch.slice(0, i);
     }
 
+    if (paymentRedirectWhiteList !== undefined) {
```

Comment:
Just a minor question, would this be preferred?
```suggestion
    if (paymentRedirectWhiteList) {
```

Reply:
I think the issue is that the API user might want to just update other fields, say `locale`, without affecting `paymentRedirectWhiteList`. In this case, what would you expect the payload for the following 4 cases would be?
1. Remove everything in `paymentRedirectWhiteList`
2. Remove one entry in `paymentRedirectWhiteList`
3. Add one entry in `paymentRedirectWhiteList`
4. Do not change `paymentRedirectWhiteList`

Reply:
May I assume the UI will get the prior settings first? If so, the field may remain previous  `paymentRedirectWhiteList` on UI for the user's further edit. And send the payload with:
1. `"paymentRedirectWhiteList": []`
2. `"paymentRedirectWhiteList": [ ... (one URL removed)]`
3. `"paymentRedirectWhiteList": [ ... (one URL added)]`
4. `"paymentRedirectWhiteList": [ ... (untouched)]`

Reply:
Since in 1. you assume empty array is delete, I think if `paymentRedirectWhiteList` is |undefined` we should leave it untouched. We should delete it if its null or empty array

Reply:
I see, so the `!== undefined` part is necessary for our scenario. But I didn't expect the null case, should I add:
```
if (paymentRedirectWhiteList === null) {
    payload.paymentRedirectWhiteList = [];
}
```
at the beginning of the function right?

Reply:
hmm i think we should add a special case and treat null as delete

Reply:
No problem, code has been updated!